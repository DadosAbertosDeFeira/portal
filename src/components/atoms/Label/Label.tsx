import type { ComponentPropsWithRef } from "react";
import { forwardRef } from "react";

export type LabelProps = ComponentPropsWithRef<"label"> & {};

export const Label = forwardRef<HTMLLabelElement, LabelProps>(function Label(
  { children, ...props },
  ref
) {
  return (
    // eslint-disable-next-line jsx-a11y/label-has-associated-control
    <label {...props} ref={ref}>
      {children}
    </label>
  );
});
