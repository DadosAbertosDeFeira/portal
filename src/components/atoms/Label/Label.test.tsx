import { render } from "@testing-library/react";

import { Label, type LabelProps } from "./Label";

describe("Label", () => {
  const makeSut = ({ ...props }: Partial<LabelProps>) => {
    return render(<Label {...props} />);
  };

  // eslint-disable-next-line jest/expect-expect
  it("renders correctly", () => {
    makeSut({});
  });
});
