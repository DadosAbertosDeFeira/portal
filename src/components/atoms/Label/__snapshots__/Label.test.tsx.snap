// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `[Function]`;
